packages/store/src/store.ts

```diff
@@ -25,8 +27,10 @@ export class Store {
   /**
    * Selects a slice of data from the store.
    */
-  select<T>(selector: (state: any, ...states: any[]) => T): Observable<T>;
-  select(selector: string | any): Observable<any>;
+  select<R, K>(selector: (state: T, ...subState: K[]) => R): Observable<R>;
+  select<R, S>(selector: (...states: S[]) => R): Observable<R>;
+  select<R, S>(memorizedSelector: (state: S) => R): Observable<R>;
+  select<R>(selector: string | StateClassStatic): Observable<R>;
```

Comment:
None of these revised selector typings give any substantial advantage over the exisitng ones as only the output type of the selector functions matters for determining the resulting Observable.